components/camel-kafka/src/main/java/org/apache/camel/component/kafka/KafkaProducer.java

```diff
@@ -189,6 +196,7 @@ public boolean hasNext() {
                     String innerTopic = msgTopic;
                     Object innerKey = null;
                     Integer innerPartitionKey = null;
+                    Long innerTimestamp = null;
```

Comment:
Should this be initialized with timeStamp? If the inner message doesn't override it use the outer message override value.

Reply:
if timestamp is provided innerTimestamp assigned with the provided timestamp else it will be null.

do you want even there is no timestamp provided it should be assigned with current timestamp?

Reply:
yeah I would also say the same, if it was explicitly provided in the inner messages, it would make sense to set it otherwise I would keep it as `null`

Reply:
In case of OVERRIDE_TOPIC , if inner message doesn't have override topic , we use the override topic from outer exchange.
I assumed we will follow the same, if overide time is not present in inner message , use the override time from outer exchange and if the override time is not set even on outer exchange, then set it as null.

Reply:
I have a mix feeling of altering this behavior, in case of topic, it would make sense to fallback to the outer topic since you need to specify a topic. However, in case of the timestamp, this is not mandatory per say to specify and thus it would be fine to have it as `null`. Also, altering the timestamp for these messages, IMHO would alter the semantics of in case you deliberately don't want to specify timestamp for some of these inner messages.

Reply:
Make sense !!!